Test CI on Windows and macOS

hugovk opened the pull request:
Follow on from https://github.com/python-twitter-tools/twitter/pull/451#issuecomment-1246747905.

~Drop support for Python 3.6, end-of-life on 2021-12-23.~ Moved to #454.

On the CI, test Ubuntu with:

* Python 2.7 (CPython and PyPy)
* the lowest and highest 3.x versions
* Any dev versions

And also test Windows on 3.8 and macOS 3.9.

boogheta commented:
Not sure to understand why coveralls fails
https://github.com/python-twitter-tools/twitter/actions/runs/3058141268/jobs/4935255053

hugovk commented:
I wonder if it's because the action isn't running on macOS and Windows? I added `if: matrix.is == 'ubuntu-latest'` because it said it was Linux only.

I've split out the 3.6 drop into https://github.com/python-twitter-tools/twitter/pull/454 to see what happens.

hugovk commented:
https://github.com/python-twitter-tools/twitter/pull/454 passed.

I've split this PR to add just Windows and macOS and it still fails.

I've reported this upstream to the action to ask for advice: https://github.com/AndreMiras/coveralls-python-action/issues/23

boogheta commented:
I merged 454, and let's wait for their answer before moving on with windows/mac then, thanks!
